app/routers/organizations.py

```diff
@@ -1,15 +1,57 @@
 from fastapi import APIRouter, status, HTTPException
 from database import client
-from models import Organization
+from models import Organization, OrganizationResponse
+import secrets
+import string
+from fastapi.encoders import jsonable_encoder
 
-router = APIRouter()
+router = APIRouter(prefix="/organizations", tags=["Organizations"])
 
 
-@router.get("/check-auth-token/{api_key}", response_model=Organization)
+@router.post("/", response_model=OrganizationResponse)
+async def create_organization(org: Organization):
+    org = jsonable_encoder(org)
+    if org["name"] is not None:
+
+        # create an API key
+        key = "".join(
+            [secrets.choice(string.ascii_letters + string.digits) for _ in range(20)]
+        )
+        number_of_loops = 3
+
+        while number_of_loops > 0:
+
+            # check if API key exists
+            if (client.quiz.organization.find_one({"key": key})) is None:
+                org["key"] = key
+                new_org = client.quiz.organization.insert_one(org)
+                created_org = client.quiz.organization.find_one(
+                    {"_id": new_org.inserted_id}
+                )
+                return created_org
+
+            else:
+                key = "".join(
+                    [
+                        secrets.choice(string.ascii_letters + string.digits)
+                        for _ in range(20)
+                    ]
+                )
+                number_of_loops -= 1
+
+        raise HTTPException(
+            status_code=500,
+            detail="Duplicate API key. Please try again.",
+        )
```

Comment:
Based on my earlier comment, you can remove this
- The response message is saying "duplicate API key..." but when the backend was asked to create an organization, no API key was passed or was required.
- the post request WILL eventually create an org with API key because of the refactored code i mentioned above. so there will never be a case where we will have duplicate API key.